app/gimlet/src/main.rs

```diff
@@ -68,47 +68,89 @@ fn system_init() {
     // you've flashed the wrong one, only.
     //
     // The revision is on pins PG[2:0], with PG2 as the MSB.
+    //
+    // In earlier firmware, we used the SP's internal pull-up resistors.
+    // However, the combination of both FPGA and SP pull-up resistors was too
+    // strong on certain hardware revisions, leading to marginal voltage
+    // readings.
+    //
+    // Instead, we reset the FPGA, which puts its pins into their default
+    // configuration (with weak pull-ups enabled).  Then, we wait for the pins
+    // to charge before reading the values.
 
-    // Un-gate the clock to GPIO bank G.
+    // Un-gate the clock to GPIO bank G (for revision ID), D (for FPGA reset),
+    // and C (for sequencer PG)
     p.RCC.ahb4enr.modify(|_, w| w.gpiogen().set_bit());
+    p.RCC.ahb4enr.modify(|_, w| w.gpioden().set_bit());
+    p.RCC.ahb4enr.modify(|_, w| w.gpiocen().set_bit());
     cortex_m::asm::dsb();
-    // PG2:0 are already inputs after reset, but without any pull resistors.
+
+    // Make PC6 (SEQ_REG_TO_SP_V3P3_PG) and PC7 (SEQ_REG_TO_SP_V1P2_PG) inputs,
+    // then wait for both of them to go high.  We time out after 1M iterations
+    // (with 100 cycles each), which is roughly 1.5s.
+    #[rustfmt::skip]
+    p.GPIOC.moder.modify(|_, w| w
+        .moder6().input()
+        .moder7().input());
```

Comment:
what was so bad about rustfmt's intended formatting? is it that the name of the bit and the `.input()` went on separate lines?

could we instead do a separate expression for each bit, so we could have the `rustfmt`-approved formatting of:
```suggestion
    p.GPIOC.moder.modify(|_, w| {
        w.moder6().input();
        w.moder7().input();
    });
```

Reply:
oh, i see that there are other `#[rustfmt::skip]`ped blocks elsewhere in this file. meh. whatever. i still would prefer to have rustfmt-approved formatting, but :woman_shrugging: